spyder/plugins/projects/utils/watcher.py

```diff
@@ -87,10 +91,28 @@ def connect_signals(self, project):
         self.event_handler.sig_file_modified.connect(project.file_modified)
 
     def start(self, workspace_folder):
-        self.observer = Observer()
-        self.observer.schedule(
-            self.event_handler, workspace_folder, recursive=True)
-        self.observer.start()
+        # Needed to handle an error caused for inotify watch limit reached.
+        # See spyder-ide/spyder#10478
+        try:
+            self.observer = Observer()
+            self.observer.schedule(
+                self.event_handler, workspace_folder, recursive=True)
+            self.observer.start()
+        except OSError:
+            QMessageBox.warning(self.parent(), "Spyder",
+                                _("An error occurred while creating "
+                                  "the watcher of changes for the working "
+                                  "directory. Please, try to increase "
+                                  "the inotify limit on your system.<br><br>"
+                                  "On Linux you can use the following "
+                                  "command:<br><br>"
+                                  "<code>sudo sysctl -n -w "
+                                  "fs.inotify.max_user_watches=524288</code>"
+                                  "<br><br>For a permanent setting of the "
+                                  "configuration you will need to edit "
+                                  "<code>/etc/sysctl.conf</code> an add "
+                                  "the following line:<br><br> <code>"
+                                  "fs.inotify.max_user_watches=524288</code>"))
```

Comment:
Let's change this message to be:

File changes for this project can't be tracked because it contains too many files. To fix this you need to increase the inotify limit in your system, with the following command:

    sudo sysctl -n -w fs.inotify.max_user_watches=524288

For a permanent solution you need to add to `/etc/sysctl.conf` the following line

    fs.inotify.max_user_watches=524288

After doing that, you need to close and start Spyder again so those changes can take effect.

Reply:
Also, please indent things like this to have more space to write the message:

```python
QMessageBox.warning(
    self.parent(),
    "Project warning",
    "File changes ..."
)

Reply:
And finally, please make `self.observer = None` above the `QMessageBox` because we won't be able to use it for this project.